k8s-charm/src/charm.py

```diff
@@ -73,6 +74,14 @@ def _on_controller_relation_changed(self, event: RelationEvent):
 
         self.unit.status = ActiveStatus()
 
+    def _on_config_changed(self, _):
+        relation = self.model.get_relation("controller")
+        if relation:
+            data = JujuDashData(relation.data[relation.app])
+            dashboard_config, nginx_config = self._render_config(data['controller_url'], data['identity_provider_url'], data['is_juju'])
+            container = self.unit.get_container("dashboard")
+            self._add_layer(container, dashboard_config, nginx_config)
```

Comment:
also.. i think add_layer is not named correctly.. because it does more than just add a pebble layer - it also pushes config files to the workload container..